src/schema/v2/me/setOrUnsetUserFlagMutation.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { GravityMutationErrorType } from "lib/gravityErrorHandler"
+import { snakeCase } from "lodash"
+import { ResolverContext } from "types/graphql"
+import { userFlagsResolver, UserFlagsType } from "./user_flags"
+
+const MutationSuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "SetOrUnsetUserFlagMutationSuccess",
+  isTypeOf: (data) => {
+    return data._type !== "GravityMutationError"
+  },
+  fields: () => ({
+    userFlags: {
```

Comment:
**question**: Instead of defining and exposing a plural userFlags type, would this feature be satisfied with a singular query field instead?

I see that many _known_ keys are already defined as an enum in Gravity [here](https://github.com/artsy/gravity/blob/main/app/models/util/user_flags_keys.rb).

```graphql
{
  me {
    userFlag(key: PURCHASED_ARTWORK_ADDED_AT) {
      key
      value
    }
  }
}
```

```graphql
mutation {
  updateUserFlag(key: PURCHASED_ARTWORK_ADDED_AT, value: null {
    key
    value
  }
}
```

Reply:
Based on [the other thread](https://github.com/artsy/metaphysics/pull/4192#discussion_r932195064), and potentially room for abuse here, we could be more restrictive with these mutations and have the backend control the value.

```graphql
mutation {
  setUserTimestampFlag(key: PURCHASED_ARTWORK_ADDED_AT) {
    key
    value
  }
}
```

```graphql
mutation {
  unsetUserTimestampFlag(key: PURCHASED_ARTWORK_ADDED_AT) {
    key
    value
  }
}
```